openreview/invitations/templates.py

```diff
@@ -232,4 +161,71 @@ def __init__(self, name, conference_id, duedate = 0,
         super(Comment, self).__init__(**self.inv_params)
 
     def add_process(self, process):
-        self.process = process.render()
\ No newline at end of file
+        self.process = process.render()
+
+def _fill_str(template_str, paper):
+    paper_params = paper.to_json()
+    pattern = '|'.join(['<{}>'.format(field) for field, value in paper_params.iteritems()])
+    matches = re.findall(pattern, template_str)
+    for match in matches:
+        discovered_field = re.sub('<|>', '', match)
+        template_str = template_str.replace(match, str(paper_params[discovered_field]))
+        print "    new value: ", template_str
+    return template_str
+
+def _fill_str_or_list(template_str_or_list, paper):
+    if type(template_str_or_list) == list:
+        return [_fill_str(v, paper) for v in template_str_or_list]
+    elif any([type(template_str_or_list) == t for t in [unicode, str]]):
+        return _fill_str(template_str_or_list, paper)
+    elif any([type(template_str_or_list) == t for t in [int, float, type(None), bool]]):
+        return template_str_or_list
+    else:
+        raise ValueError('first argument must be list or string: ', value)
+
+def _fill_template(template, paper, exclude=[]):
+    new_template = {}
+    for field, value in template.iteritems():
+        if type(value) != dict:
+            new_template[field] = _fill_str_or_list(value, paper)
+        elif field not in exclude:
+            # recursion
+            new_template[field] = _fill_template(value, paper)
+        else:
+            new_template[field] = value
+    return new_template
+
+def from_template(invitation_template, paper):
+    '''
+    web, process, duedate, cdate, rdate, ddate, multiReply, and taskCompletionCount
+    should not be updated.
+    '''
+
+    exclusions = [
+        'web','process','duedate','cdate','rdate','ddate',
+        'multiReply','taskCompletionCount'
+    ]
+
+    new_params = _fill_template(invitation_template, paper, exclude = exclusions)
+
+    return openreview.Invitation(
```

Comment:
@melisabok is this what you had in mind?

Reply:
I really like it! It is simpler to understand